src/baseline_bots/bots/smart_order_accepter_bot.py

```diff
@@ -143,6 +201,26 @@ def is_support_for_selected_orders(self, support_order: str) -> bool:
             return True
         return False
 
+    def is_support_for_given_orders(self, support_order, orders):
+        """Determine if selected support order for neighbour corresponds to given list of orders"""
+        order_tokens = get_order_tokens(support_order)
+        if order_tokens[2].split()[1] not in orders:
+            return True  # it's okay to support other power than allies
+        selected_order = get_order_tokens(orders[order_tokens[2].split()[1]])
+
+        if (
```

Comment:
add comments throughout.

Additionally, you may want to turn the auto formatter off in some sections

Reply:
I didn't write this part so I think it is best for somebody else to add comments and type hints. I can make an issue if you'd like.